script.js

```diff
@@ -1,12 +1,243 @@
+
+  return code;
+}
+
+//episode
+function createEpisodeElement(episodeData) {
+  let episode = document.createElement("div");
+
+  episode.className = "episodeItem";
+  episode.innerHTML = `
+    <h2>${episodeData.name} - ${padSeasonAndEpisode(episodeData)}</h2>
+    <div class="image">
+      <img src=${episodeData.image.medium} alt='${episodeData.season} ${episodeData.name
+    }'/>
+    </div>
+    <div class="description">
+      ${episodeData.summary}
+    </div>
+    
+    `;
+
+  return episode;
+}
+
+//header
+function createHeader() {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let header = document.createElement("div");
+  header.className = "header";
+  rootElem.appendChild(header);
+  let title = document.createElement("h1");
+  header.appendChild(title);
+  title.innerHTML = "TV show project";
+}
+
+//section (with episodes)
+function createSection() {
+  const rootElem = document.getElementById("root");
+  let section = document.createElement("div");
+  section.className = "section";
+  rootElem.appendChild(section);
+  return section;
+}
+//tvmaze
+
+function createMazeDiv() {
+  const header = document.querySelector(".header");
+  let maze = document.createElement("div");
+  maze.className = "mazeDiv";
+  header.appendChild(maze);
+  maze.innerHTML = `Originally, data was obtained from <a href = 'https://www.tvmaze.com/' target= '_blank'>TVMaze.com</a>`;
+  return maze;
 }
 
+//search
+function createSearchBar() {
+  let form = document.createElement("form");
+  form.innerHTML = `<input type ='search' class = 'search' placeholder = 'your search term...'/><span class = 'searchOutput'> <span>Displaying 73/73 episodes</span> 
+  
+  </span>`;
+  return form;
+}
+
+//select
+function createSelect() {
+  const header = document.querySelector(".header");
+  let select = document.createElement("select");
+  select.className = "select";
+  header.appendChild(select);
+  // console.log(header);
+}
+
+// write a function which creates a popup episode
+function createPopUp(episodeId) {
+  episodeId = parseInt(episodeId)
+  const popUp = document.createElement("div");
+  popUp.id = "popupEpisode";
+  popUp.className = "episodeItem";
+
+  // 1. find the episode with this id
+  const allEpisodes = episodes;
+  const episodeData = allEpisodes.find((episode) => episode.id === episodeId);
+
+  popUp.innerHTML = `
+    <button id="popupCloseBtn">X</button>
+    <h2>${episodeData.name} - ${padSeasonAndEpisode(episodeData)}</h2>
+    <div class="image">
+      <img src=${episodeData.image.medium} alt='${episodeData.season} ${episodeData.name
+    }'/>
+    </div>
+    <div class="description">
+      ${episodeData.summary}
+    </div>
+  `;
+
+  // button for closing the popUp:
+  //get the button and add event listener
+  let closeBtn = popUp.children[0];
+  closeBtn.addEventListener("click", (e) => {
+    // remove the popup from the DOM
+    popUp.remove();
+  });
+
```

Comment:
Cool idea to put the button to close and return to all episodes here. very creative ad nicely coded, quite easy to understand what you were doing here. The attention to detail you have is also very apparent.